src/sentry/integrations/vsts/issues.py

```diff
@@ -86,3 +88,45 @@ def get_issue(self, issue_id, **kwargs):
             'title': work_item['fields']['System.Title'],
             'description': work_item['fields'].get('System.Description')
         }
+
+    def sync_assignee_outbound(self, external_issue, user, assign=True, **kwargs):
+        client = self.get_client()
+        assignee = None
+
+        # TODO(LB): What's the scope here? is this correct?
+        # Get a list of all users in a given scope. How do we define scope?
+        # https://docs.microsoft.com/en-us/rest/api/vsts/graph/users/list?view=vsts-rest-4.1
+
+        if assign is True:
+            vsts_users = client.get_users(self.model.name)
+            sentry_emails = [email.email.lower() for email in user.get_verified_emails()]
+
+            for vsts_user in vsts_users['value']:
+                vsts_email = vsts_user.get(u'mailAddress')
+                if vsts_email and vsts_email.lower() in sentry_emails:
+                    assignee = vsts_user['mailAddress']
+                    break
+
+            if assignee is None:
+                self.logger.info(
+                    'vsts.assignee-not-found',
+                    extra={
+                        'integration_id': external_issue.integration_id,
+                        'user_id': user.id,
+                        'issue_key': external_issue.key,
+                    }
+                )
+                return
+
+        try:
+            client.update_work_item(
+                self.instance, external_issue.key, assigned_to=assignee)
```

Comment:
so can you assign just based on email address? what happens if you try to assign to an invalid email address?

i guess i'm wondering if the above fetching of users is even necessary and if you could just try to assign to each of the user's verified emails. if the api actually tells you that it failed to assign, that could save a few requests?

for jira in order to assign, i needed to have the username, so fetching users was necessary. not a big deal if there's a reason to do it here too, just figured i'd mention

Reply:
@macqueen I'm not sure what happens when you use an invalid email address and/or an address that is not associated with an account that makes sense. I can experiment with it, but this entire behavior is not documented for vsts. I realized it could be done by trying. So, I'll give it another go.

Reply:
yeah not much was documented for jira either unfortunately, so i think trial and error is the best way to figure this out probably

Reply:
@macqueen When I started implementing this idea, I'm not sure I agree with this. A user can have any number of verified email accounts. So, the number of api calls possible is not capped. Whereas asking for all the users in the scope and checking will just require 1 extra api call guaranteed.  I assume most users have 2 or less verified email addresses... But I'd need to look. I can follow up with another PR when I get a chance or we can talk about it when you get back.